pkg/controllers/managementuser/clusterauthtoken/common/user.go

```diff
@@ -21,16 +23,45 @@ func NewClusterAuthToken(token *managementv3.Token, hashedValue string) (*cluste
 		TypeMeta: metav1.TypeMeta{
 			Kind: "ClusterAuthToken",
 		},
-		UserName:      token.UserID,
-		SecretKeyHash: hashedValue,
-		ExpiresAt:     token.ExpiresAt,
-		Enabled:       tokenEnabled,
+		UserName:  token.UserID,
+		ExpiresAt: token.ExpiresAt,
+		Enabled:   tokenEnabled,
 	}
 	return result, nil
 }
 
-// VerifyClusterAuthToken verifies that a provided secret key is valid for the given clusterAuthToken.
-func VerifyClusterAuthToken(secretKey string, clusterAuthToken *clusterv3.ClusterAuthToken) error {
+// NewClusterAuthSecret creates a new secret from the given token and its hash value
+// The cluster auth token is managed separately.
+// Does not create the secret in the remote cluster.
+func NewClusterAuthSecret(token *managementv3.Token, hashedValue string) (*corev1.Secret, error) {
+	result := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: ClusterAuthSecretName(token.ObjectMeta.Name),
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Secret",
+		},
+		Data: map[string][]byte{
+			"hash": []byte(hashedValue),
+		},
+	}
+	return result, nil
+}
+
+// ClusterAuthSecretName computes the name of the cluster auth secret for a
+// token T from the name of T.
+func ClusterAuthSecretName(tokenName string) string {
+	return "cat-" + tokenName
```

Comment:
Q: Can we just use the token name?

Reply:
note, even when switching to the token name alone I would prefer to keep the function. as the unchanging API, concealing the possibly changing internal. go should be able to auto-inline this.

Reply:
the function is used by kube-api-auth

Reply:
commit a3fadb08124149384773e593dd598f380244253a